packages/@aws-cdk-containers/ecs-service-extensions/lib/extensions/publisher.ts

```diff
@@ -0,0 +1,146 @@
+import * as ecs from '@aws-cdk/aws-ecs';
+import * as sns from '@aws-cdk/aws-sns';
+import { Service } from '../service';
+import { Container } from './container';
+import { ContainerMutatingHook, ServiceExtension } from './extension-interfaces';
+
+// Keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from '@aws-cdk/core';
+
+/**
+ * An interface that will be implemented by all the resources that can be published events to.
+ */
+export interface IPublisher {
+  publish(taskDefinition: ecs.TaskDefinition): void;
+
+  environmentVar(): { [key: string]: string };
+}
```

Comment:
The name of of the extension will be: `InjecterExtension`.

and it will take as input `injectables: []Injectable`
Then we can loop through each injectable and do this:
```ts
for (const injectable of injectables) {
  if (injectable as GrantInjectable).grant !== undefined {
     injectable.grant(taskDef)
  }
  merge(this.environment, injectable.environmentVariables())
}
```

Reply:
Let me know what y'all think!

Reply:
That makes a lot of sense to me! I think overall the concern here is: when we design a new interface, we need to think it as an interface instead of a concrete class.

Reply:
> SNS topics can implement this interface, S3 buckets, or DynamoDB tables for example.
In the future, to add RDS for example as an injected we can create another interface ConnectInjectable.

Yup, this sounds super cool ❤️